src/ansi-c/ansi_c_entry_point.cpp

```diff
@@ -248,7 +248,10 @@ bool generate_ansi_c_start_function(
     {
       // ok
     }
-    else if(parameters.size()==2 || parameters.size()==3)
+    else if(
+      parameters.size() >= 2 && parameters[1].type().id() == ID_pointer &&
```

Comment:
Did you mean `parameters[0]`?

Reply:
No, `parameters[0]` is what we'd expect to be `argc` if I' not mistaken?

Reply:
The one place that enforces the order is the C standard (and any related architecture descriptions). The user, however, may supply arbitrary (syntactically valid) C code, even one that does not respect the calling conventions set out in the C standard. If the user does supply such code (as is done in the accompanying regression test), then we can only tell them that they got it wrong, which is what we do via the error message in the `else` branch of this code. The change being proposed here should make sure that we do end up in this `else` branch.

Does this address your concern? Should I still be making changes to the implementation? Should I perhaps add the above as a comment?

Reply:
Yes, I'm happy to approve. The detail in your response might be nice to have in the comments though since this is not immediately clear.

Reply:
Comment added to the code.

Reply:
The comment is very helpful.